### PR TITLE
Feature/equals hashcode tostring for archive models

### PR DESCRIPTION
**Pull request by liujoshua**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,12 +6,12 @@
...
     <groupId>org.sagebionetworks</groupId>
     <artifactId>BridgeDataUploadUtils</artifactId>
-    <version>0.2.1</version>
+    <version>0.2.3</version>
 
     <properties>
         <aws.version>1.10.49</aws.version>
         <jackson.version>2.7.4</jackson.version>
-        <java.version>1.8</java.version>
+        <java.version>1.7</java.version>
```
Comment (zaclinsage) [inline pom.xml]: did you mingle this pull with the multidex pull codes?

File: src/main/java/org/sagebionetworks/bridge/data/ArchiveInfo.java [hunk 2]
```diff
@@ -25,6 +25,29 @@
             this.filename = filename;
             this.timestamp = timestamp;
         }
+
+        @Override public boolean equals(Object o) {
```
Comment (zaclinsage) [inline src/main/java/org/sagebionetworks/bridge/data/ArchiveInfo.java]: i thought we already have equals, hashcode and tostring in ArchiveInfo class? Did you remove it from previous commits?

File: pom.xml (hunk 1 above)
Comment (liujoshua) [inline pom.xml]: Yes, I forgot the JDK change hadn't been merged, so it has the same commits. Should be fixed now that I've merged the JDK PR

File: src/main/java/org/sagebionetworks/bridge/data/ArchiveInfo.java (hunk 2 above)
Comment (liujoshua) [inline src/main/java/org/sagebionetworks/bridge/data/ArchiveInfo.java]: I thought I had added it too. Not sure why, but it disappeared.

**Pull request merged**